File: code/pheidi-app/src/app/week.ts
import { Day } from './day'
import { DistanceType } from './distance-type.enum';

export class Week {
    weekNumber: number;
    lastLongRun: boolean;

    distances: Record<DistanceType, number> = {
        [DistanceType.Long]: 0,
        [DistanceType.Half]: 0,
        [DistanceType.Quarter]: 0,
        [DistanceType.QuarterUp]: 0,
        [DistanceType.None]: 0
    };
    taper: boolean;

    constructor(weekNumber: number) {
        this.weekNumber = weekNumber;
    }
}

File: code/Pheidi/Pheidi.Common/Week.cs
namespace Pheidi.Common;

public class Week(int weekNumber)
{
    public int WeekNumber { get; } = weekNumber;

    public bool LastLongRun { get; set; }

    public Dictionary<DistanceType, decimal> Distances { get; private set; } = new()
    {
        [DistanceType.Half] = 0,
        [DistanceType.Long] = 0,
        [DistanceType.None] = 0,
        [DistanceType.Quarter] = 0
    };

    public bool Taper { get; internal set; }

    public override string ToString() => $"{WeekNumber} + {Distances[DistanceType.Long]}";
}
